src/Network/Network/VirtualNetwork/Subnet/ResetAzureVirtualNetworkSubnetPolicyCommand.cs

```diff
@@ -0,0 +1,107 @@
+            HelpMessage = "Name of the service deletion. Use Get-AzureRmAvailableServiceDelegation command to list of ServiceName")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = "SetByResourceId",
+            HelpMessage = "Name of the service deletion. Use Get-AzureRmAvailableServiceDelegation command to list of ServiceName")]
+        [ValidateNotNullOrEmpty]
+        public string ServiceName { get; set; }
+
+        [Alias("InputObject")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = "DefaultParameterSet",
+            ValueFromPipeline = true,
+            HelpMessage = "The virtualNetwork")]
+        public PSVirtualNetwork VirtualNetwork { get; set; }
+
+        [Parameter(
+            ParameterSetName = "SetByResourceId",
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource Id of the virtual network resource")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            if (this.IsParameterBound(c => c.ResourceId) || !string.IsNullOrWhiteSpace(this.ResourceId))
+            {
+                var resourceIdentifier = new ResourceIdentifier(this.ResourceId.Trim('"'));
+                var resourceGroupName = resourceIdentifier.ResourceGroupName;
+                var virtualNetworkName = resourceIdentifier.ResourceName;
+                this.VirtualNetwork = GetVirtualNetwork(resourceGroupName, virtualNetworkName);
+            }
+
+            // Verify if the subnet exists in the VirtualNetwork
+            var subnet = this.VirtualNetwork.Subnets.SingleOrDefault(resource => string.Equals(resource.Name, this.Name, StringComparison.CurrentCultureIgnoreCase));
+            if (subnet == null)
+            {
+                throw new ArgumentException("Subnet with the specified name does not exist");
+            }
+
+            if(!this.ServiceName.Contains("/"))
+            {
+                throw new ArgumentException("ServiceName should be like Microsoft.Sql/managedInstances");
+            }
+            
+            if (ShouldProcess($"Subnet {this.Name}", "Reset"))
+            {
+                // call prepareNetworkPolicies API
+                var unPrepareRequestParams = new UnprepareNetworkPoliciesRequest(this.ServiceName);
+                this.NetworkClient.NetworkManagementClient.Subnets.UnprepareNetworkPolicies(this.VirtualNetwork.ResourceGroupName,
+                                this.VirtualNetwork.Name, this.Name, unPrepareRequestParams);
+
+                var getVirtualNetwork = GetVirtualNetwork(this.VirtualNetwork.ResourceGroupName, this.VirtualNetwork.Name);
+                var getSubnet = getVirtualNetwork.Subnets.SingleOrDefault(resource => string.Equals(resource.Name, this.Name, StringComparison.CurrentCultureIgnoreCase));
+
```

Comment:
null check?

Reply:
this is the same subnet we just updated, so it can't be null. If there are any network errors then call throws exception so we are good.